espressif/vscode-esp-idf-extension
# Filter serial devices by default and allow user to disable chip id in serial port list

## Description

Add `idf.enableSerialPortChipIdRequest` and `idf.useSerialPortVendorProductFilter` which are set true by default.

Now the serial ports listed with the **ESP-IDF: Select port to use** command will only show know USB serial ports by filtering them with productID and vendorID.  This can be disabled with `"idf.useSerialPortVendorProductFilter": false` in settings.json

esptool.py chip_id is executed when running the previous command to get the chip name and show it in the list. This can be disabled with `"idf.enableSerialPortChipIdRequest": false` in settings.json.

Add `idf.usbSerialPortFilters` for user to define the list of USB PID VID filters.

Fixes #1274

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## Steps to test this pull request

Provide a list of steps to test changes in this PR and required output

1. Click on "**ESP-IDF: Select port to use** ". Now only Espressif devices should be shown by default.
2. Set `"idf.useSerialPortVendorProductFilter": false` and Click on "**ESP-IDF: Select port to use** ". Now all serial ports will be shown. If set true, the behavior is back as step 1.
3. Set `"idf.enableSerialPortChipIdRequest": false` and Click on "**ESP-IDF: Select port to use** ".  Now the esptool.py chip_id is not executed and the chip name is not shown in list. If set true, the behavior is same as step 1.

- Expected behaviour:

- Expected output:

## How has this been tested?

Manual testing with steps above.

**Test Configuration**:
* ESP-IDF Version: 5.2.2
* OS (Windows,Linux and macOS): macOS

## Checklist
